src/lib/y2partitioner/widgets/format_and_mount.rb

```diff
@@ -177,143 +230,183 @@ def mount?
 
     # BlkDevice Filesystem selector
     class BlkDeviceFilesystem < CWM::ComboBox
-      def initialize(filesystem)
+      def initialize(options)
         textdomain "storage"
 
-        @filesystem = filesystem
+        @options = options
+      end
+
+      def opt
+        %i(hstretch notify)
       end
 
       def init
-        self.value = @filesystem
+        fs_type = @options.filesystem_type
+        self.value = fs_type ? fs_type.to_sym : nil
       end
 
       def label
         _("Filesystem")
       end
 
       def items
-        Y2Storage::Filesystems::Type.all.map do |fs|
-          [fs.to_s, fs.to_human_string]
+        return [] unless @options.filesystem_type
+
+        Y2Storage::Filesystems::Type.all.select { |fs| supported?(fs) }.map do |fs|
+          [fs.to_sym, fs.to_human_string]
         end
       end
 
+      def supported?(fs)
+        %i(swap btrfs ext2 ext3 ext4 vfat xfs reiserfs).include?(fs.to_sym)
```

Comment:
constant and to be honest I do not expect it in widget, but more likely in Partition Type class.